---
title: "GeneExpression"
author: "Antonio Pano"
date: "2022-11-28"
output: html_document
---


Genome-wide DNA methylation was assayed in a cohort of opioid-naïve participants who underwent standard dental surgery followed by opioid self-administration. Saliva samples were collected longitudinally before surgery (visit 1), and at two postsurgery visits at 2.7 ± 1.5 days (visit 2), and 39 ± 10 days (visit 3) after the discontinuation of opioid analgesics.


```{r setup, include=FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE) 
```

```{r}
library(R.utils)
library(tidyverse)
library(Seurat)
library(DESeq2)
```






##### Read Counts

```{r}
gunzip("GSE117872_good_Data_TPM.txt.gz")
gunzip("GSE117872_good_Data_cellinfo.txt.gz")

drug <- read.table("GSE117872_good_Data_TPM.txt")
drug_info <- read.table("GSE117872_good_Data_cellinfo.txt")
```



# Gene Expression

Can't tell from head() which samples were treated and which weren't.
```{r}
head(drug)
```
```{r}
head(drug_info)
```


2: Constructing a DESeqDataSet Object
```{r}
dds <- DESeqDataSetFromMatrix(countData = round(drug),
                       colData = drug_info,
                       design = ~ drug_status)
```

3. Pre-filtering
Removing rows with low gene counts
Keeping rows that have at least 10 reads total
```{r}
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]
```

Note: Collapse technical replicates. Never collapse biological replicates.

4. Setting factor level & running DESeq. Saving results.
```{r}
relevel(dds$drug_status, ref = "Resistant")

dds <- DESeq(dds)
res <- results(dds)
```

Exploring Results
* Adjusting p-value to 0.01
```{r}
res
summary(res)

res0.01 <- results(dds, alpha = 0.01)
summary(res0.01)
```
Contrasts
```{r}
resultsNames(dds)
```
E.g. resistant, sensitive, holiday
COME BACK
```{r}
results(dds, contrast = c("Resistant", "Sensitive", "Holiday"))
```

# MA Plot
```{r}
plotMA(res)
```
